Makefile

```diff
@@ -17,14 +22,18 @@ check-lint: check-tabs
 		-not -name \*.swp \
 		| xargs shellcheck --severity=warning && echo Success!
 
-check-system: $(MICROOVN_SNAP)
-	echo "Running functional tests";					\
-	$(CURDIR)/.bats/bats-core/bin/bats tests/
+$(ALL_TESTS): $(MICROOVN_SNAP)
+	echo "Running functional test $@";					\
+	$(CURDIR)/.bats/bats-core/bin/bats $@
+
+check-system: $(ALL_TESTS)
 
-$(MICROOVN_SNAP):
+$(MICROOVN_SNAP): $(MICROOVN_SOURCES) $(SNAP_SOURCES) $(COMMAND_WRAPPERS)
 	echo "Building the snap";						\
 	snapcraft pack -v -o $(MICROOVN_SNAP)
 
 clean:
 	rm -f $(MICROOVN_SNAP_PATH);						\
 	snapcraft clean
+
+.PHONY: $(ALL_TESTS) clean check-system check-lint check-tabs
```

Comment:
Do we really want to run system tests when just running `make`? I think I'd expect a plain `make` to only build the snap.

Reply:
I may misunderstand something about `.PHONY`, but my intention was ensure that all tests in `$(ALL_TESTS)` are always executed. Since the test targets (recipes?) are now real files, `Make` could (and I saw it happen in my testing) assume that the target is "up-to-date" and not execute it.

As for adding `clean`, `check-system` and others, I included them out of habit. I think it's better to have these "virtual" targets  explicitly in the `.PHONY` on the off chance that a file will be created with the same name sometime in the future and will cause unexpected behavior of `make`.

> Do we really want to run system tests when just running make?

This is already happening with current implementation. Since the `check-system` target does not represent any real file, it's always "out-of-date" so plain `make` will re-execute it. I'll look into how to fix this behavior (I'm happy to take any suggestions, if you have some.).

Reply:
> Do we really want to run system tests when just running make?

I added `.DEFAULT_GOAL` so that when user runs `make`, it only builds the snap without running tests.